mesonbuild/coredata.py

```diff
@@ -34,14 +34,14 @@
 
 if T.TYPE_CHECKING:
     from . import dependencies
-    from .compilers.compilers import Compiler
+    from .compilers.compilers import Compiler, CompileResult
     from .environment import Environment
     from .mesonlib import OptionOverrideProxy, FileOrString
     from .cmake.traceparser import CMakeCacheEntry
 
     OptionDictType = T.Union[T.Dict[str, 'UserOption[T.Any]'], OptionOverrideProxy]
     KeyedOptionDictType = T.Union[T.Dict['OptionKey', 'UserOption[T.Any]'], OptionOverrideProxy]
-    CompilerCheckCacheKey = T.Tuple[T.Tuple[str, ...], str, 'FileOrString', T.Tuple[str, ...], str]
+    CompilerCheckCacheKey = T.Tuple[T.Tuple[str, ...], str, FileOrString, T.Tuple[str, ...], str]
```

Comment:
Incidentally, this FileOrString is only imported in T.TYPE_CHECKING, and it is also only *used* in T.TYPE_CHECKING. So I figured there was zero point in quoting it, and then flake8 detected it as used, which it is. The other cases were a bit harder since they weren't already inside T.TYPE_CHECKING...